core/src/main/scala/akka/kafka/scaladsl/Committer.scala

```diff
@@ -26,20 +27,20 @@ object Committer {
    * Batches offsets and commits them to Kafka, emits `CommittableOffsetBatch` for every committed batch.
    */
   def batchFlow(settings: CommitterSettings): Flow[Committable, CommittableOffsetBatch, NotUsed] = {
-    val offsetBatches = Flow[Committable]
-      .groupedWeightedWithin(settings.maxBatch, settings.maxInterval)(_.batchSize)
-      .map(CommittableOffsetBatch.apply)
+    val offsetBatches: Flow[Committable, CommittableOffsetBatch, NotUsed] =
+      Flow
+        .fromGraph(new BatchingFlowStage(settings))
+
     // See https://github.com/akka/alpakka-kafka/issues/882
     import akka.kafka.CommitDelivery._
     settings.delivery match {
       case WaitForAck =>
         offsetBatches
-          .mapAsyncUnordered(settings.parallelism) { b =>
-            b.commitInternal().map(_ => b)(ExecutionContexts.sameThreadExecutionContext)
+          .mapAsyncUnordered(settings.parallelism) { batch =>
+            batch.commitInternal().map(_ => batch)(ExecutionContexts.sameThreadExecutionContext)
           }
       case SendAndForget =>
-        offsetBatches
-          .map(_.tellCommit())
+        offsetBatches.map(_.tellCommit())
```

Comment:
This is built-in now and should not be needed here anymore.
But the `CommitDelivery` distinction is missing in the stage implementation.

Reply:
> This is built-in now and should not be needed here anymore.

Committing is still used for normal scenario's, stage only commits on failures. Can you elaborate what is not needed?

> But the `CommitDelivery` distinction is missing in the stage implementation.

Indeed, adding!

Reply:
> This is built-in now and should not be needed here anymore.

@ennru We discussed that the normal committing logic could remain as a separate `mapAsyncUnordered` operator step so similar logic wouldn't need to be implemented in the custom graph stage.  WDYT?

Reply:
Yes, makes sense. And we do not need the distinction with tell commit I guess. As everything will fail just after anyway, so just using tell commits would be the right thing to do?

Reply:
What about failed commits? Do we get stream failures with `tellCommit`?

Reply:
No, but if the commit is triggered by a stream failure, we shouldn't override that failure anyway. Just try our best to get the offset to be committed over to the Kafka broker, no matter what happens after that.

Reply:
> No, but if the commit is triggered by a stream failure, we shouldn't override that failure anyway. Just try our best to get the offset to be committed over to the Kafka broker, no matter what happens after that.

That sounds logical. 

May I propose to use only `tellCommit` in the `CommitCollectorStage` (thus ignore the `CommitDelivery` setting. And keep it as it is for the regular flow (i.e. respecting the setting in the `Committer`)? 
We might still want to get feedback on commit failures in normal scenario's (i.e. when no upstream failure occurred). With `tellCommit` no stream restart possible in case of kafka commit failures.